platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/lifecycle/MigrationContext.java

```diff
@@ -62,7 +62,7 @@ public interface MigrationContext<V> {
      * Returns the startup networks in use.
      */
     @NonNull
-    StartupNetworks startupNetworks();
+    <N> StartupNetworks<N> startupNetworks();
```

Comment:
I find this technique of creating generic types to remove dependencies from HAPI has a smell to it.

Using the `MigrationContext` outside of HederaApp feels weird anyway. Are schemas and schema migration used outside of HederaApp? I think it is worth considering moving this functionality out of the State API (in a separate module or into HederaApp) or redesigning it to make its reuse more natural.